canvas/src/layer/layer_2d.rs

```diff
@@ -3,32 +3,169 @@ use crate::{
 };
 pub use glyph_brush::{GlyphCruncher, HorizontalAlign, Layout, VerticalAlign};
 pub use lyon::{
-  path::{builder::PathBuilder, traits::PathIterator, Path, Winding},
+  geom::Arc,
+  math::{point, rect},
+  path::{builder::PathBuilder, path::Builder, traits::PathIterator, EndpointId, Path, Winding},
   tessellation::*,
 };
+
 use std::{
   cmp::PartialEq,
   ops::{Deref, DerefMut},
 };
 
+pub type Vector = euclid::default::Vector2D<f32>;
+pub type Angle = euclid::Angle<f32>;
+
 /// The 2d layer is a two-dimensional grid. The coordinate (0, 0) is at the
 /// upper-left corner of the canvas. Along the X-axis, values increase towards
 /// the right edge of the canvas. Along the Y-axis, values increase towards the
 /// bottom edge of the canvas.
-#[derive(Default, Debug, Clone)]
+// #[derive(Default, Debug, Clone)]
 pub struct Rendering2DLayer<'a> {
   state_stack: Vec<State>,
   pub(crate) commands: Vec<Command<'a>>,
+  path: Option<Builder>,
+}
+
+impl<'a> Default for Rendering2DLayer<'a> {
+  fn default() -> Self { Self::new() }
 }
 
 impl<'a> Rendering2DLayer<'a> {
   pub fn new() -> Self {
     Self {
       state_stack: vec![State::new()],
       commands: vec![],
+      path: None,
     }
   }
 
+  #[inline]
```

Comment:
please define private method in the tail, let user can always see the public method first.